code/game/objects/items/devices/radio/beacon.dm

```diff
@@ -1,63 +1,69 @@
-/obj/item/radio/beacon
-	name = "Tracking Beacon"
+/obj/item/beacon
+	name = "tracking beacon"
 	desc = "A beacon used by a teleporter."
+	icon = 'icons/obj/radio.dmi'
 	icon_state = "beacon"
 	item_state = "signaler"
 	origin_tech = "bluespace=1"
+	flags = CONDUCT
+	slot_flags = SLOT_FLAG_BELT
+	throw_speed = 2
+	throw_range = 9
+	w_class = WEIGHT_CLASS_SMALL
+
+	materials = list(MAT_METAL = 200, MAT_GLASS = 100)
 	var/syndicate = FALSE
 	var/area_bypass = FALSE
 	var/cc_beacon = FALSE //set if allowed to teleport to even if on zlevel2
 
-/obj/item/radio/beacon/New()
+/obj/item/beacon/New()
 	..()
-	GLOB.beacons += src
+	GLOB.beacons |= src
 
-/obj/item/radio/beacon/Destroy()
+/obj/item/beacon/Destroy()
 	GLOB.beacons -= src
 	return ..()
 
-/obj/item/radio/beacon/emag_act(user as mob)
+/obj/item/beacon/emag_act(user as mob)
 	if(!emagged)
 		emagged = TRUE
 		syndicate = TRUE
 		to_chat(user, "<span class='notice'>The This beacon now only be locked on to by emagged teleporters!</span>")
 		return TRUE
 
-/obj/item/radio/beacon/hear_talk()
-	return
 
 /// Probably a better way of doing this, I'm lazy.
-/obj/item/radio/beacon/bacon
+/obj/item/beacon/bacon
 
-/obj/item/radio/beacon/bacon/proc/digest_delay()
+/obj/item/beacon/bacon/proc/digest_delay()
 	QDEL_IN(src, 600)
 
-/obj/item/radio/beacon/emagged
+/obj/item/beacon/emagged
 	syndicate = TRUE
 	emagged = TRUE
 
 // SINGULO BEACON SPAWNER
-/obj/item/radio/beacon/syndicate
+/obj/item/beacon/syndicate
 	name = "suspicious beacon"
 	desc = "A label on it reads: <i>Activate to have a singularity beacon teleported to your location</i>."
 	origin_tech = "bluespace=6;syndicate=5"
 	syndicate = TRUE
 	var/obj/machinery/computer/syndicate_depot/teleporter/mycomputer
```

Comment:
This variable seems to be unused apart from the `Destroy()`, and my compiler doesn't catch any undefined vars if I remove this

Reply:
Out of Scope